src/csharp/Grpc.Tools.Tests/scripts/fakeprotoc.py

```diff
@@ -220,32 +220,43 @@ def _get_cs_files_to_generate(protofile, proto_to_generated):
     return cs_files_to_generate
 
 
-def _generate_cs_files(protofile, cs_files_to_generate, out_dir, projectdir):
+def _generate_cs_files(protofile, cs_files_to_generate, grpc_out_dir,
+                       csharp_out_dir, projectdir):
     """Create expected cs files."""
     _write_debug("\ngenerate_cs_files")
 
     if not cs_files_to_generate:
         _write_debug("No .cs files matching proto file name %s" % protofile)
         return
 
-    if not os.path.isabs(out_dir):
+    if not os.path.isabs(grpc_out_dir):
         # if not absolute, it is relative to project directory
-        out_dir = os.path.abspath("%s/%s" % (projectdir, out_dir))
+        grpc_out_dir = os.path.abspath("%s/%s" % (projectdir, grpc_out_dir))
 
-    # Ensure out_dir exists
-    if not os.path.isdir(out_dir):
-        os.makedirs(out_dir)
+    if not os.path.isabs(csharp_out_dir):
+        # if not absolute, it is relative to project directory
+        grpc_out_dir = os.path.abspath("%s/%s" % (projectdir, csharp_out_dir))
```

Comment:
bug here: need to assign to `csharp_out_dir =`?

Reply:
Fixed